### PR TITLE
[CI] Run short integration tests on PR

### PR DESCRIPTION
**Pull request by uenoku**
Considering that we break integration tests and revert commits frequently, it is better to run short integration tests on PR CI to catch test failures in integration tests. This PR changes to run short integration tests on pull requests.

### CODE REVIEW

**Comment by uenoku** [comment]
I'm just a bit concerned that running ccache on every PR might break something.

And it's not related to this PR but can we use ccache for normal tests on CI?

File: .github/workflows/shortIntegrationTests.yml [hunk 1]
```diff
@@ -13,6 +13,8 @@ on:
   push:
     branches:
       - main
+  pull_request:
+    types: [assigned, opened, synchronize, reopened]
```
Comment (teqdruid) [inline .github/workflows/shortIntegrationTests.yml:17]: 'assigned' doesn't seem right.

**Review by teqdruid — APPROVED** [review]
Yeah, given the recent frequency at which the integration tests have been broken this seems appropriate.

**Comment by teqdruid** [comment]
> And it's not related to this PR but can we use ccache for normal tests on CI?

The "normal" tests don't run a unified build, so we cache the results of the MLIR build separately from the CIRCT build. Since the CIRCT build doesn't take long, I don't think it would add much value but would increase the complexity.

**Comment by uenoku** [comment]
> The "normal" tests don't run a unified build, so we cache the results of the MLIR build separately from the CIRCT build. Since the CIRCT build doesn't take long, I don't think it would add much value but would increase the complexity.

Thanks. It makes sense.

**Pull request merged**